File: plugins/org.eclipse.osee.orcs.rest/src/org/eclipse/osee/orcs/rest/internal/search/artifact/ArtifactSearch.java
/*******************************************************************************
 * Copyright (c) 2012 Boeing.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 *
 * Contributors:
 *     Boeing - initial API and implementation
 *******************************************************************************/
package org.eclipse.osee.orcs.rest.internal.search.artifact;

import javax.ws.rs.core.Request;
import javax.ws.rs.core.UriInfo;

/**
 * @author John R. Misinco
 * @author Roberto E. Escobar
 */
public abstract class ArtifactSearch {
   private final UriInfo uriInfo;
   private final Request request;

   public ArtifactSearch(UriInfo uriInfo, Request request) {
      this.uriInfo = uriInfo;
      this.request = request;
   }

   public UriInfo getUriInfo() {
      return uriInfo;
   }

   public Request getRequest() {
      return request;
   }
}

File: plugins/org.eclipse.osee.web/src/app/ple/artifact-explorer/lib/types/artifact-search.ts
/*********************************************************************
 * Copyright (c) 2024 Boeing
 *
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *     Boeing - initial API and implementation
 **********************************************************************/
import { NamedId } from '@osee/shared/types';

export type AdvancedSearchCriteria = {
	artifactTypes: NamedId[];
	attributeTypes: NamedId[];
	exactMatch: boolean;
	searchById: boolean;
};

export const defaultAdvancedSearchCriteria: AdvancedSearchCriteria = {
	artifactTypes: [],
	attributeTypes: [],
	exactMatch: false,
	searchById: false,
};
